app/src/test/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModelTest.kt

```diff
@@ -138,5 +186,49 @@ class LoginSSOViewModelTest {
         loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginSSOError.DialogError.GenericError::class
         (loginViewModel.loginState.loginSSOError as LoginSSOError.DialogError.GenericError).coreFailure shouldBe networkFailure
     }
+
+    @Test
+    fun `given successful sso result, when establish the sso login session, then SSOLoginResult is passed`(){
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Success(authSession)
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        runTest { loginViewModel.establishSSOSession(DeepLinkResult.SSOLogin.Success("","")) }
+
+        coVerify(exactly = 1) { loginViewModel.navigateToConvScreen() }
+    }
+
+    @Test
+    fun `given invalid cookie sso result, when establish the sso login session, then SSOLoginResult fails`(){
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Failure.InvalidCookie
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        runTest { loginViewModel.establishSSOSession(DeepLinkResult.SSOLogin.Success("","")) }
+
+        coVerify(exactly = 0) { loginViewModel.navigateToConvScreen() }
+    }
```

Comment:
We should assert as well registering client is not invoked ?

Reply:
Yes, I'm adding this failures as well. But for passing the cookie, actually we need the server Id as well to fetch the serverConfig [at the moment is hardcoded at line 68 - there are 2 reasons for that, one is the sso now only working on staging, and the other reason is we need some updates on the serverConfig repository side that is not updated yet.